x-pack/plugins/ml/server/models/results_service/results_service.ts

```diff
@@ -708,6 +721,35 @@ export function resultsServiceProvider(mlClient: MlClient, client?: IScopedClust
       finalResults.bucketResults.push([timestamp, eventCount]);
     });
 
+    const { getAnnotations } = annotationServiceProvider(client!);
+    const resp = await getAnnotations({
+      jobIds: [jobId],
+      earliestMs: start,
+      latestMs: end,
+      maxAnnotations: 1000,
+    });
+
+    const annotationResults = resp.annotations[jobId] || [];
+    annotationResults.forEach((annotation) => {
+      const timestamp = Number(annotation?.timestamp);
+      const endTimestamp = Number(annotation?.end_timestamp);
+      if (timestamp === endTimestamp) {
+        finalResults.annotationResultsLine.push({
+          dataValue: timestamp,
+          details: annotation.annotation,
+          header: dateFormatter(timestamp),
```

Comment:
I think you will need to move the time formatting part to the client, as currently all the times are being formatted in the timezone of the server, and so are not consistent with the timezone configured in `dateFormat:tz` under custom settings.

Reply:
Good catch - updated in 3f2c45c807b4eb637c8e3c7e220203c328128f39